### DISCUSSION TITLE
Saving added markers/lines/areas

### DISCUSSION THREAD

**Discussion by egeres** [opengeos/leafmap#168] [Q&A] [answered]
It's possible to add makers, lines and polygonal areas directly in the notebook's ipywidget. **Is it also possible to save these changes from that UI**?

If not, in which attribute is it possible to access this information in a `leafmap.Map` object? 

![image](https://user-images.githubusercontent.com/8045792/148684623-f136cb1a-9cff-4e77-a3cd-899634fc2f99.png)

**Comment by giswqs** [comment]
Do you mean saving the shapes you draw on the map?

**Comment by egeres** [comment]
Yes!

**Reply by giswqs** [reply]
It is possible. I will implement it.

**Comment by giswqs** [comment, marked as answer]
This feature has been implemented. See #170 

- `m.user_roi` to retrieve the last draw feature
- `m.user_rois` to retrieve all draw features as a list
- `m.save_draw_features(out_file)` to save draw features as a GeoJSON file.
